src/main/java/org/folio/service/orders/flows/update/unopen/UnOpenCompositeOrderManager.java

```diff
@@ -189,17 +185,18 @@ private Future<Void> processInventoryOnlyWithItems(CompositePoLine compPOL, Requ
   }
 
   private Future<List<Piece>> deleteExpectedPieces(CompositePoLine compPOL, RequestContext requestContext) {
-    if (!PoLineCommonUtil.isReceiptNotRequired(compPOL.getReceiptStatus()) && Boolean.FALSE.equals(compPOL.getCheckinItems())) {
-      return pieceStorageService.getExpectedPiecesByLineId(compPOL.getId(), requestContext)
-        .compose(pieceCollection -> {
-          if (isNotEmpty(pieceCollection.getPieces())) {
-            return pieceStorageService.deletePiecesByIds(pieceCollection.getPieces().stream().map(Piece::getId).collect(toList()), requestContext)
-                                .map(v -> pieceCollection.getPieces());
-          }
-          return Future.succeededFuture(Collections.emptyList());
-        });
+    if (!PoLineCommonUtil.isItemsUpdateRequired(compPOL)) {
```

Comment:
changed to more generic check here to be the same in other places.
Also for example in UI when creating POL with ReceiptStatus = 'Receipt not required status', the receiving workflow becomes Independent automatically, so these 2 checks are replaced to simplify the code.
Also before this change ReceiptStatus always was Pending because of the issue with order of operations described earlier.